plugins/generator-1.21.1/neoforge-1.21.1/procedures/utils/projectiles/arrow.java.ftl

```diff
@@ -0,0 +1,13 @@
+private static AbstractArrow initArrowProjectile(AbstractArrow entityToSpawn, Entity shooter, float damage,
+		boolean silent, boolean fire, boolean particles, AbstractArrow.Pickup pickup) {

```

Comment:
The constructor of `AbstractArrow` sets projectile's custom name from `firedFromWeapon` if it is passed. This is the only side effect of its invocation we don't "overwrite" by other method calls, but we can do that too with these lines:
```java
	if (entityToSpawn.hasCustomName())
		entityToSpawn.setCustomName(null);
```
But I just thought that a custom projectile (maybe not from MCreator mod) may be using `setCustomName()` in its own constructor, so not sure if it makes sense to add these 2 lines at all?

Reply:
Is this name even visible anywhere?

Reply:
It shouldn't be without the call to `setCustomNameVisible(true)` that parent constructor doesn't perform.